frontend-vue3/src/composables/api/profile.ts

```diff
@@ -0,0 +1,50 @@
+import { useAxios, type RequestOptions } from "@/composables/api/axios";
+import { toRefs } from "vue";
+
+interface ProfileQueryParams {
+  query?: string;
+  page: number;
+}
+
+export function useProfileAPI() {
+  /**
+   * Composable function for making requests to the profile API
+   *
+   * @returns - An object containing reactive state of the request and helper functions for API requests
+   */
+
+  const baseUrl = "/users/";
+  const { state, get, postForm, put, detailUrl, searchUrl } = useAxios(baseUrl);
+
+  async function search(params: ProfileQueryParams) {
+    return get(searchUrl(params));
+  }
+
+  async function retrieve(id: string | number) {
+    return get(detailUrl(id));
+  }
+
+  async function update(id: string | number, data: any, options?: RequestOptions) {
+    return put(detailUrl(id), data, options);
+  }
+
+  async function uploadProfilePicture(id: string | number, file: any) {
+    const formData = new FormData();
+    formData.append("file", file);
+    return postForm(`${detailUrl(id)}upload_picture/`, formData, {
```

Comment:
maybe useAxios can also expose url building functions that hang off things like `detailUrl` - something like `buildUrl(detailUrl(id), 'upload_picture')` or an optional second argument path for a signature like `detailUrl(id, 'upload_picture')`